rust/src/value.rs

```diff
@@ -39,12 +41,20 @@ impl Value {
         )
     }
 
+    /// Try to decode the value as base64 binary data,
+    /// otherwise return UTF-8 string.
+    pub fn from_possibly_base64_encoded(value: String) -> Self {
+        base64::engine::general_purpose::STANDARD
+            .decode(&value)
+            .map_or(Self::Utf8(value), Value::Binary)
+    }
+
     /// Read data from given filepath.
     ///
     /// Supports both UTF-8 and non-UTF-8 contents.
     pub fn from_path(path: String) -> Result<Self, VaultError> {
         if let Ok(content) = std::fs::read_to_string(&path) {
-            Ok(Self::Utf8(content))
+            Ok(Self::from_possibly_base64_encoded(content))
```

Comment:
This needs to depend on the operation. Now a base64 encoded secret gets converted to binary and using a binary blob will now change default behavior

Reply:
Yes, changed the approach and now only the direct encrypt and decrypt will use base64.